### PR TITLE
ts/ikh-indicator

### PR DESCRIPTION
**Pull request by pawellysy**

### CODE REVIEW

File: js/Stock/Indicators/IKHIndicator.js [hunk 1]
```diff
@@ -6,12 +6,26 @@
  *
```
Comment (bre1470) [inline js/Stock/Indicators/IKHIndicator.js:6]: This whole file needs to be deleted and with that the referencing masters files needs to be updated.

File: ts/Stock/Indicators/IKH/IKHIndicator.ts [hunk 2]
```diff
@@ -0,0 +1,960 @@
...
+const color = Color.parse;
+import H from '../../../Core/Globals.js';
+import U from '../../../Core/Utilities.js';
+import {
+    IKHDrawSenkouSpanObject,
+    IKHGapExtensionObject,
+    IKHOptions,
+    IKHParamsOptions,
+    IKHSenkouSpanOptions
+} from './IKHOptions';
```
Comment (bre1470) [inline ts/Stock/Indicators/IKH/IKHIndicator.ts:38]: The import should make it clear that it is about types. Therefor explicitly do `import type`.

File: ts/Stock/Indicators/IKH/IKHIndicator.ts [hunk 3]
```diff
@@ -0,0 +1,960 @@
...
+    *  Properties
+    *
+    * */
+
+    public data: Array<IKHPoint> = void 0 as any;
+    public options: IKHOptions = void 0 as any;
+    public points: Array<IKHPoint> = void 0 as any;
+    public pointValKey = 'tenkanSen';
+    public nameComponents = ['periodSenkouSpanB', 'period', 'periodTenkan'];
+
```
Comment (bre1470) [inline ts/Stock/Indicators/IKH/IKHIndicator.ts:374]: The last two properties should be kept in the extend call for the class prototype.

File: ts/Stock/Indicators/IKH/IKHIndicator.ts [hunk 4]
```diff
@@ -0,0 +1,960 @@
...
+    ): IndicatorValuesObject<TLinkedSeries> | undefined;
+    graphCollection: Array<string>;
+    graphsenkouSpan: SVGElement | undefined;
+    ikhMap: Highcharts.Dictionary<Array<IKHPoint>>;
+    nextPoints?: Array<IKHPoint>;
+    options: IKHOptions;
+    pointValKey: string;
+    pointClass: typeof IKHPoint;
+    points: Array<IKHPoint>;
+    pointArrayMap: Array<keyof IKHPoint>;
```
Comment (bre1470) [inline ts/Stock/Indicators/IKH/IKHIndicator.ts:926]: Every property here, that will be not part of the extend call (has a default value) needs to be moved to class pattern and in some cases initialized with `void 0 as any`. Exception to this rule is the pointClass property, which stays in the interface.

**Review by bre1470 — CHANGES_REQUESTED** [review]
Thank you! Please see comments for requested changes.

File: js/Stock/Indicators/IKHIndicator.js (hunk 1 above)
Comment (pawellysy) [reply js/Stock/Indicators/IKHIndicator.js:6]: Deleted file, and updated masters files

File: ts/Stock/Indicators/IKH/IKHIndicator.ts (hunk 2 above)
Comment (pawellysy) [reply ts/Stock/Indicators/IKH/IKHIndicator.ts]: Added 'type' import

File: ts/Stock/Indicators/IKH/IKHIndicator.ts (hunk 3 above)
Comment (pawellysy) [reply ts/Stock/Indicators/IKH/IKHIndicator.ts]: Moved those two properties to extend call

File: ts/Stock/Indicators/IKH/IKHIndicator.ts (hunk 4 above)
Comment (pawellysy) [reply ts/Stock/Indicators/IKH/IKHIndicator.ts]: Moved given properties to class pattern

**Review by bre1470 — APPROVED** [review]
Thank you! Looks good. 👍

**Comment by sebastianbochan** [comment]
I had to re-run tests (disconnect issue) and now, the indicator is ready to merge.

**Pull request merged**